src/bigml/api/utils.clj

```diff
@@ -5,20 +5,65 @@
 (ns bigml.api.utils
   "Offers functions specific for BigML clusters.
      https://bigml.com/developers/clusters"
+  (:require [clojure.set :as set])
   (:require (bigml.api [core :as api])))
 
+(defn resource-type
+  "Returns the resource type of the specified resource.
+   The resource can be specified through:
+    - its full UUID, e.g., 'model/12234567890';
+    - a minimal map-representation, e.g., { :resource '...' };
+    - a full JSON representation."
+  [resource]
+  (let [resource (or (:resource resource) (str resource))]
+    (first (clojure.string/split resource #"/"))))
+
+(defn- if-contains
+  "Returns the passed map m if it contains the given keypath ks,
+  otherwise nil. A keypath is a sequence of keys for recursive lookup."
+  [m ks]
+  (and (get-in m ks) m))
+
+(defn normalized-resource
+  "Makes sure the passed resource contains enough information for input
+   normalization (i.e., :input_fields and additionally :restype/field
+   when by-name is true. If it does not, then the resoure is fetched."
+  [resource by-name]
+  (let [ res-type (resource-type resource)]
+   (or (and (if-contains resource [:input_fields])
```

Comment:
Actually, `if-contains` has a slightly different semantics in that it returns the identified value, if present. I will rename it to `get-keypath`.

On the other hand, I could you `get-in` instead of `if-contains`/`get-keypath` but I somehow appreciate expressing the idea that the two forms (`(if-contains resource [:input_fields])` and `(if-contains resource [(keyword res-type) :fields])`) have the same semantics, i.e., accessing the field list -- only in two different places (`: input_fields` or `[(keyword res-type) :fields]`) depending on the JSON layout.

If you do not see this value, I am happy to use `get-in` instead...